src/Common/tests/System/Net/Capability.Sockets.cs

```diff
@@ -47,5 +47,10 @@ public static bool IPv4Support()
         {
             return Socket.OSSupportsIPv4;
         }
+
+        public static bool UnixDomainSocketsSupport()
+        {
+            return !RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
```

Comment:
Unfortunately you can't add new APIs easily (you'd have to change corefx for dnx, netnative, netcoreforcoreclr and Desktop).

You could add the code in Capabilities since one of the uses of this code is as documentation/samples.